### PR TITLE
cloud: add new stanza to specify custom certs for Prometheus

### PR DESCRIPTION
**Pull request by taroface**
Add a stanza to the Prometheus manifest that allows the user to specify custom certs. The `cockroach-node` specifies the default node secret used by the K8s Operator, as advised by @chrisseto.

Relates to https://github.com/cockroachdb/cockroach-operator/issues/469.

A doc update to the Prometheus tutorial will refer to this change.

### CODE REVIEW

File: cloud/kubernetes/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -60,11 +60,12 @@ spec:
...
-      # certificate authority.
-      caFile: "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
-      # This overrides the hostname verification check for the admin
-      # UI port to match our quickstart secure-mode cluster setup.
+      ca:
+        secret:
+          key: ca.crt
+          # This is the secret name used by the CockroachDB Kubernetes Operator.
+          # When using a custom CA, replace this with your secret name
+          name: cockroachdb-node
```
Comment (chrisseto) [inline cloud/kubernetes/prometheus/prometheus.yaml:68]: @keith-mcclellan, is this something that we want to support in the long run or should we wait for the operator to have a stand alone ca.crt secret?

**Review by chrisseto — APPROVED** [review]
LGTM pending a quick answer from Keith.

I believe our link to the Prometheus operator (`https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.20/bundle.yaml`) on [this page](https://github.com/cockroachdb/cockroach/tree/4673b25d1de5eadf5c1ccf6966b6265f5b116a19/cloud/kubernetes/prometheus#monitoring) is now outdated.

I used https://github.com/prometheus-operator/prometheus-operator/blob/v0.47.1/bundle.yaml when testing the configuration. Would you mind updating that link as well? 😄

File: cloud/kubernetes/prometheus/prometheus.yaml (hunk 1 above)
Comment (keith-mcclellan) [reply cloud/kubernetes/prometheus/prometheus.yaml:68]: This is an easy fix and will work indefinitely (we aren't removing the ca.crt from cockroachdb-node, just making a standalone secret that also hosts the file) so I'm okay with it.  We can't get that work done before the GA release of the operator anyway.

**Comment by keith-mcclellan** [comment]
> LGTM pending a quick answer from Keith.
> 
> I believe our link to the Prometheus operator (`https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.20/bundle.yaml`) on [this page](https://github.com/cockroachdb/cockroach/tree/4673b25d1de5eadf5c1ccf6966b6265f5b116a19/cloud/kubernetes/prometheus#monitoring) is now outdated.
> 
> I used https://github.com/prometheus-operator/prometheus-operator/blob/v0.47.1/bundle.yaml when testing the configuration. Would you mind updating that link as well? 😄

I agree with Chris that we should update the docs to point to the latest stable version of prometheus.  Is there a way for us to trigger a refresh of this periodically?

**Comment by taroface** [comment]
> I agree with Chris that we should update the docs to point to the latest stable version of prometheus. Is there a way for us to trigger a refresh of this periodically?

I'm not sure. Currently our Prometheus doc on the CRL site says:

> If you run into an error when installing the Prometheus Operator, first try updating the [release version](https://github.com/prometheus-operator/prometheus-operator/blob/master/RELEASE.md) specified in the below command and reapplying the manifest. If this does not work, please [file an issue](file-an-issue.html).

Which isn't ideal. Perhaps we could add a step before the bundle install that asks people to check that Prometheus releases page first? (Incidentally it looks like the next version is scheduled for tomorrow.)

**Comment by taroface** [comment]
@keith-mcclellan @chrisseto Does the extra instruction in that latest commit work for you both?

**Review by chrisseto — APPROVED** [review]
LGTM!

**Pull request merged**